apps/api/v2/src/ee/event-types/event-types_2024_06_14/services/input-event-types.service.ts

```diff
@@ -114,7 +227,245 @@ export class InputEventTypesService_2024_06_14 {
+  }
+
+  transformInputConfirmationPolicy(
+    requiresConfirmation: CreateEventTypeInput_2024_06_14["confirmationPolicy"]
+  ) {
+    return transformConfirmationPolicyApiToInternal(requiresConfirmation);
+  }
   transformInputRecurrignEvent(recurrence: CreateEventTypeInput_2024_06_14["recurrence"]) {
     return transformRecurrenceApiToInternal(recurrence);
   }
+
+  transformInputEventTypeColor(color: CreateEventTypeInput_2024_06_14["color"]) {
+    return transformEventColorsApiToInternal(color);
+  }
+
+  transformInputSeatOptions(seats: CreateEventTypeInput_2024_06_14["seats"]) {
+    return transformSeatsApiToInternal(seats);
+  }
+  async validateEventTypeInputs(
+    eventTypeId?: number,
+    seatsEnabled?: boolean,
+    locations?: ReturnType<typeof this.transformInputLocations>,
+    requiresConfirmation?: boolean,
+    eventName?: string
+  ) {
+    // Retrieve event type details from the database if eventTypeId is provided
+
+    if (eventTypeId != null) {
+      const eventTypeDb = await this.eventTypesRepository.getEventTypeWithSeats(eventTypeId);
+      const seatsEnabledDb = !!eventTypeDb?.seatsPerTimeSlot && eventTypeDb.seatsPerTimeSlot > 0;
+      const locationsDb = this.outputEventTypesService.transformLocations(eventTypeDb?.locations);
+      const requiresConfirmationDb = eventTypeDb?.requiresConfirmation ?? false;
+
+      seatsEnabled &&
+        (await this.validateSeatsInput(
+          locations,
+          requiresConfirmation,
+          locationsDb,
+          requiresConfirmationDb,
+          true
+        ));
+      locations &&
+        locations.length > 1 &&
+        (await this.validateLocationsInput(seatsEnabled, locations, seatsEnabledDb, true));
+      requiresConfirmation &&
+        (await this.validateRequiresConfirmationInput(
+          requiresConfirmation,
+          seatsEnabled,
+          seatsEnabledDb,
+          true
+        ));
+      eventName && (await this.validateCustomEventNameInput(eventName));
+    } else {
+      seatsEnabled &&
+        (await this.validateSeatsInput(locations, requiresConfirmation, undefined, undefined, false));
+      locations?.length > 1 && (await this.validateLocationsInput(seatsEnabled, locations, undefined, false));
+      requiresConfirmation &&
+        (await this.validateRequiresConfirmationInput(requiresConfirmation, seatsEnabled, undefined, false));
+      eventName && (await this.validateCustomEventNameInput(eventName));
+    }
+  }
+  async validateSeatsInput(
+    locations?: ReturnType<typeof this.transformInputLocations>,
+    requiresConfirmation?: boolean,
+    locationsDb?: ReturnType<typeof this.transformInputLocations>,
+    requiresConfirmationDb?: boolean,
+    validateDb = false
+  ) {
+    if (validateDb) {
+      // Validate based on both provided and database values
+
+      // 1. Check if provided locations have more than one entry
+      if (locations && locations.length > 1) {
```

Comment:
Why are locations and seats connected / related ?

Reply:
ok this i now think about it makes sense that seats event has to have 1 location